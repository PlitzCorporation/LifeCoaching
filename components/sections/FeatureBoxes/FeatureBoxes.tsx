import altCreator from '@/helpers/altCreator';
import { ExtraBoxesProps } from '@/types/pages';
import Image from 'next/image';
import React from 'react';

export interface FeatureBoxesProps {
	boxesData: ExtraBoxesProps[];
}

const FeatureBoxes = (props: FeatureBoxesProps) => {
	return (
		<section className="relative z-10 -mt-8 mb-9">
			<div className="grid grid-cols-12 gap-8 lg:gap-12">
				{props.boxesData.map((box) => (
					<div
						key={box._id}
						className="col-span-12 lg:col-span-6 border-t-10 border-plitz-primary"
					>
						<div className="bg-white py-9 lg:py-12 px-14 lg:px-18 shadow-plitz-box-4-32 h-full">
							<Image
								src={`${process.env.NEXT_PUBLIC_CMS}/${box.eimg}`}
								alt={'icon ' + altCreator(`${box.eimg}`)}
								width={37}
								height={37}
								className="mb-6"
							/>
							<h3 className="font-bold text-primary text-balance">
								{box.etitle}
							</h3>
							<p className="mt-4 whitespace-pre-wrap text-balance">
								{box.econtent}
							</p>
						</div>
					</div>
				))}
			</div>
		</section>
	);
};

export default FeatureBoxes;
