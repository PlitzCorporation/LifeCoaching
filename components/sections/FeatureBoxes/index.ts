import FeatureBoxes from './FeatureBoxes';

export default FeatureBoxes;
