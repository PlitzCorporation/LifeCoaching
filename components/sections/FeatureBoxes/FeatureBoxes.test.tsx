import { render } from '@testing-library/react';
import FeatureBoxes, {
	FeatureBoxesProps,
} from '@/components/sections/FeatureBoxes/FeatureBoxes';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="false" />;
	},
}));

describe('FeatureBoxes', () => {
	const OLD_ENV = process.env;

	beforeAll(() => {
		jest.resetModules();
		process.env.NEXT_PUBLIC_CMS = 'https://thedavid.plitz7.com';
	});

	afterAll(() => {
		process.env = OLD_ENV;
	});

	const boxesData: FeatureBoxesProps['boxesData'] = [
		{
			_id: '1',
			etitle: 'Box 1',
			esubtitle: 'featuredBox',
			econtent: 'Description 1',
			eposition: 1,
			eimg: 'uploads/Reason4Hope/image.jpg',
		},
		{
			_id: '2',
			etitle: 'Box 2',
			esubtitle: 'featuredBox',
			econtent: 'Description 2',
			eposition: 2,
			eimg: 'uploads/Reason4Hope/image.jpg',
		},
	];

	it('renders the correct number of boxes', () => {
		const { container } = render(<FeatureBoxes boxesData={boxesData} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
