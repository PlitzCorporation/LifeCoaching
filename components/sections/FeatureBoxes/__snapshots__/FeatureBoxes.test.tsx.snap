// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureBoxes renders the correct number of boxes 1`] = `
<section
  class="plitz-container relative z-10 -mt-4 lg:-mt-8 mb-9"
>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-8 lg:gap-12"
  >
    <div
      class="lg:col-span-6 border-t-10 border-plitz-primary"
    >
      <div
        class="bg-white py-9 lg:py-12 px-14 lg:px-18 shadow-plitz-box-4-32 h-full"
      >
        <img
          alt="icon uploads/Reason4Hope/image"
          class="mb-6"
          height="37"
          priority="false"
          src="https://thedavid.plitz7.com/uploads/Reason4Hope/image.jpg"
          width="37"
        />
        <h3
          class="font-bold text-primary text-balance"
        >
          Box 1
        </h3>
        <p
          class="mt-4 whitespace-pre-wrap text-balance"
        >
          Description 1
        </p>
      </div>
    </div>
    <div
      class="lg:col-span-6 border-t-10 border-plitz-primary"
    >
      <div
        class="bg-white py-9 lg:py-12 px-14 lg:px-18 shadow-plitz-box-4-32 h-full"
      >
        <img
          alt="icon uploads/Reason4Hope/image"
          class="mb-6"
          height="37"
          priority="false"
          src="https://thedavid.plitz7.com/uploads/Reason4Hope/image.jpg"
          width="37"
        />
        <h3
          class="font-bold text-primary text-balance"
        >
          Box 2
        </h3>
        <p
          class="mt-4 whitespace-pre-wrap text-balance"
        >
          Description 2
        </p>
      </div>
    </div>
  </div>
</section>
`;
